File: pdu/python/std_msgs/pdu_cdr_conv_UInt32.py
from .pdu_pytype_UInt32 import UInt32
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_UInt32(writer: CdrWriter, src: UInt32):
    writer.write_uint32(src.data)


def cdr_body_to_py_UInt32(reader: CdrReader, dst: UInt32):
    dst.data = reader.read_uint32()
    return dst


def py_to_cdr_UInt32(src: UInt32) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_UInt32(writer, src)
    return writer.bytes()


def cdr_to_py_UInt32(cdr_payload) -> UInt32:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = UInt32()
    return cdr_body_to_py_UInt32(reader, dst)

File: pdu/javascript/std_msgs/pdu_cdr_conv_UInt32.js
import { UInt32 } from './pdu_jstype_UInt32.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduUInt32Converter {
    /**
     * @param {PduCdrWriter} writer
     * @param { UInt32 } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint32(src.data);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { UInt32 } dst
     * @returns { UInt32 }
     */
    static cdr_body_to_js(reader, dst) {
        dst.data = reader.read_uint32();
        return dst;
    }

    /**
     * @param { UInt32 } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { UInt32 }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new UInt32());
    }
}
